[EWL-4388] Migrate Lead News Block

chelsiejohnston opened the pull request:
## Ticket(s)

**Jira Ticket**

- [EWL-4388: Migrate Lead News Block](https://issues.ama-assn.org/browse/EWL-4388)


## Description

Adds the Lead News pattern to SG2. 
This includes article stub and it's variant article stub with small image


## To Test
- [ ] http://localhost:3000/?p=organisms-lead-news
- [ ] Observe lead news with an article stub and article with small image

...

## Visual Regressions
none yet
...

## Relevant Screenshots/GIFs
<img width="1439" alt="screen shot 2018-01-26 at 12 12 52 pm" src="https://user-images.githubusercontent.com/2271747/35454099-46b82eea-0292-11e8-8b91-5717de448069.png">

...

## Remaining Tasks
none
...


## Additional Notes
The D8 side does not pull in this organism. Instead is recreates this organism with D8 overrides so it can override the various molecules as needed.
...

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] added pattern files

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

csj1012 committed:
[EWL-4388] add pattern to the Topic page pattern and start populating .json data

designerneil commented:
@chelsiejohnston doesn't the lead news pull in the article preview molecule?

designerneil committed:
Updates topic preview styling and video sizes

designerneil committed:
removes article preview molecule
updates article stub molecule to match article preview
changes article stub to help D8 out when consuming it

designerneil committed:
updates classes to ama__article-stub__metadata

designerneil committed:
Updates made to article preview per PR https://github.com/AmericanMedicalAssociation/ama-style-guide-2/pull/61

designerneil committed:
Adds article-stub related markdown

designerneil committed:
Adds article small variant to lead news

designerneil committed:
Adds article small variant to lead news

designerneil committed:
Merge branch 'feature/EWL-4388-migrate-lead-news-block' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-4388-migrate-lead-news-block

designerneil committed:
Adds lead news markdown

designerneil committed:
Adds lead news markdown

Commit:
update naming for as variant

designerneil committed:
removes extraneous article stub as related markdown

designerneil committed:
renames article stub variants with as variant

Commit:
Merge pull request #61 from AmericanMedicalAssociation/feature/EWL-4192-Migrate-Article-Preview

EWL-4192 Article Preview

labbydev reviewed (changes requested):
It looks like it is missing the version using the "Hero - No media" view mode, and an stub showing the "Related article - No image" view mode. Also the image for the small image one is on the right when it should be on the left.

designerneil committed:
Merge branch 'feature/EWL-4388-migrate-lead-news-block' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-4388-migrate-lead-news-block

designerneil committed:
Fixes lead news per PR
- Adds related content article stub
- modifies title only
- modifies metadata to use a parent container
- changes some of the conditionals for article stub

designerneil commented:
@labbydev you're absolutely right. I was missing some of the patterns for lead news. Those missing patterns have been put into the organism. I also had make changes to the article stub molecules to accommodate the lead news organism. Thanks for your thorough review.

labbydev commented on styleguide/source/_patterns/03-organisms/lead-news.twig:
```diff
@@ -0,0 +1,7 @@
+    {% set articleStub = article %}
+    {% include "@molecules/article-stub.twig" %}
+  {% endfor %}
+</div>
```
empty line

labbydev commented on styleguide/source/assets/scss/02-molecules/_article-metadata.scss:
```diff
@@ -1,12 +1,6 @@
-.ama__topic_article-preview_metadata {
+.ama__article-stub__metadata {
```
Seems like this should probably live in `_article-stub.scss` instead of having another partial for the metadata

designerneil replied:
the article stub metadata has it's own css so it has parity with the twig file called article-metadata.twig

designerneil replied:
Because our css gets balled into one. I don't think having the meta css is going to be an issue as files sizes go. I rather have separate files per twig file if that makes sense.

labbydev replied:
@designerneil I think the issue here is the classname. If the class name is `ama__article-stub__metadata` therefore saying "this thing belongs to `article-stub`" then it should be included within the `_article-stub.scss`. If you want to keep the `scss` files to align with the `twig` files then that's fine, but the class name should also be updated to relfect that. Therefore the class should change to `ama__article-metadata` in that case (since that's the name of the pattern. This would no longer make it a child element of the `article-stub`.

chelsiejohnston replied:
popping in to say that in addition to the above a class of `ama__article-metadata` should be added as a top-level wrapper to `article-metadata.twig`, and any includes of that pattern should adjusted accordingly. Then, styling for the article metadata pattern can go in `_article-metadata.scss`.

designerneil replied:
The reason why the metadata doesn't have a wrapper is because D8 is consuming the date and read time as two separate entities: 
field--node--field-publish-date--news-article.html.twig 
field--node--field-read-time--news-article.html.twig

without a parent container that contains both the metadata entities

labbydev commented on styleguide/source/assets/scss/02-molecules/_article-stub.scss:
```diff
@@ -0,0 +1,71 @@
+.ama__article-stub {
+  @include gutter($margin-bottom-half...);
+
+  &--container-image {
```
Is this a sub element or a modifier? Seems like it is probably a sub element.

designerneil replied:
good quesiton

designerneil replied:
i think it's neither a modifer or sub element. but just a name

labbydev commented on styleguide/source/_patterns/01-atoms/video.twig:
```diff
@@ -1,3 +1,3 @@
-<div class="ama__video-container">
+<div class="ama__video__container">
   <iframe src="{{ video.src }}" frameborder="0" class="iframe-video"></iframe>
 </div>
```
empty line

designerneil committed:
updates per PR suggestion

designerneil commented:
empty lines added but github is not showing them

labbydev commented:
See additional comments around thread about that `_article-metada.scss`

designerneil committed:
changes names of article-metadata to better reflect BEM standards for what the molecule is

designerneil commented:
The name has been changes to ama__article-metadata
